rfcs/proposed/core_types/README.md

```diff
@@ -253,6 +284,40 @@ The test infrastructure could generate all possible core type combinations using
 | **Runtime impact** | Negligible compared to task scheduling overhead |
 | **Affinity operations** | Linear in number of core types, performed once at arena creation |
 
+## Alternatives Considered
+
+### Alternative 1: Accept Multiple Constraints Instances
+
+Instead of modifying the `constraints` struct, introduce a new `task_arena` constructor that accepts a vector of
+`constraints` instances. The arena would compute the union of affinity masks from all provided constraints, enabling
+specification of multiple NUMA nodes and core types in a single arena.
+
+```cpp
+// Example usage
+tbb::task_arena arena({
+    tbb::task_arena::constraints{}.set_core_type(core_types[1]),
+    tbb::task_arena::constraints{}.set_core_type(core_types[2])
+});
+```
+
+**Pros:**
+- More scalable: can extend to any other constraint type and specify multiple platform portions as a unified constraint
+- Reuses existing `constraints` struct without modification
+- Avoids bit-packing, format markers, and special value handling
+- No risk of misinterpretation of existing single core type constraints
+
+**Cons:**
+- Requires creating multiple `constraints` objects for simple core type combinations
+- Vector of `constraints` instances vs. single integer field with bit-packing creates memory overhead
+- Unclear how to handle conflicting `max_concurrency` or `max_threads_per_core` across instances
```

Comment:
A possible way of handling conflicting `max_concurrency` or `max_threads_per_core` can be interpreting each constraint instance as an independent constraint with its own `max_concurrency` and `max_threads_per_core`. Once affinity mask is determined for each constraint, the union of them is created, so the sum of `max_concurrency`-ies, which can be used the overall `max_concurrency` for result `task_arena` instance.

`max_threads_per_core` is not needed to be handled anyhow specifically as it is directly reflected in the mask of the constraint.

Reply:
I asked a similar question in [PR1926](https://github.com/uxlfoundation/oneTBB/pull/1926) but it applies here too. Assume there is a system with three core types, indexes 0, 1 and 2 and there are 4 of each kind. What if the user wants 6 slots and to use the most performant cores. From what you describe, I think they would create a constraint for index 2 and max_concurrency of 4 slots. And then another for index 1 and max_concurrency of 2?  Is that right?  The result would be a max_concurrency of six and a mask that includes both index 2 and index 1 core types?  So if all core types are idle, the arena may populate with 4 index 1 types and 2 index 2 types, or 2 index 1 types and 4 index 2 types, right?  Would one be expected over the other?

Reply:
I have removed this from the cons. If we adopt this alternative, the new behavior will be part of the proposal.

Reply:
But you would need to pick which e-cores are in the mask.